contracts/token/ERC20/extensions/ERC20Freezable.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.26;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {IERC7943} from "../../../interfaces/IERC7943.sol";
+
+/**
+ * @dev Extension of {ERC20} that allows to implement a freezing
+ * mechanism that can be managed by an authorized account with the
+ * {_freezeTokens} and {_unfreezeTokens} functions.
+ *
+ * The freezing mechanism provides the guarantee to the contract owner
+ * (e.g. a DAO or a well-configured multisig) that a specific amount
+ * of tokens held by an account won't be transferable until those
+ * tokens are unfrozen using {_unfreezeTokens}.
+ */
+abstract contract ERC20Freezable is ERC20 {
+    /// @dev Frozen amount of tokens per address.
+    mapping(address account => uint256) private _frozenBalances;
+
+    /// @dev The operation failed because the user has insufficient unfrozen balance.
+    error ERC20InsufficientUnfrozenBalance(address user, uint256 needed, uint256 available);
```

Comment:
What about `ERC7943InsufficientUnfrozenBalance` ? Is not mandatory but just curious of your thoughts